docs/docs/tutorials/providers.md

```diff
@@ -0,0 +1,469 @@
+2. Install the MetaMask extension for your browser. You can download MetaMask from their website: https://metamask.io/.
+3. Create a new HTML file in your code editor.
+4. Copy and paste the following code into your HTML file:
+
+    ```html
+    <!DOCTYPE html>
+    <html lang="en">
+    	<head>
+    		<meta charset="UTF-8" />
+    		<title>Connecting to the Ethereum network with Web3.js and MetaMask</title>
+    	</head>
+    	<body>
+    		<h1>Connecting to the Ethereum network with Web3.js and MetaMask</h1>
+    		<div id="log">
+    			You need to approve connecting this website to MetaMask. Click on the MetaMask icon
+    			in the browser extension, if it did not show a popup already.
+    		</div>
+
+    		<script src="https://cdn.jsdelivr.net/npm/web3@4.0.1-rc.1/dist/web3.min.js"></script>
+    		<script>
+    			window.addEventListener('load', function () {
+    				// Check if web3 is available
+    				if (typeof window.ethereum !== 'undefined') {
+    					// Use the browser-injected Ethereum provider
+    					web3 = new Web3(window.ethereum);
+    					// Request access to the user's MetaMask account
+    					window.ethereum.enable();
+    					// Get the user's accounts
+    					web3.eth.getAccounts().then(function (accounts) {
+    						// Show the first account
+    						document.getElementById('log').innerHTML =
+    							'Connected with MetaMask account: ' + accounts[0];
+    					});
+    				} else {
+    					// If web3 is not available, provide instructions to install MetaMask
+    					document.getElementById('log').innerHTML =
+    						'Please install MetaMask to connect to the Ethereum network.';
+    				}
+    			});
+    		</script>
+    	</body>
+    </html>
+    ```
+
+    5. Save the file as `index.html`.
+    6. Open a terminal or command prompt window.
+    7. Navigate to the directory where your HTML file is located.
+    8. Type `npm init -y` and press Enter. This will create a `package.json` file in the current directory.
+    9. Type `npm install --save express` and press Enter. This will install the Express module and add it to your project's dependencies.
+    10. Create a new file called `server.js` in the same directory as your HTML file.
+    11. Copy and paste the following code into `server.js`:
+
+    ```js
+    const express = require('express');
+    const app = express();
+    const path = require('path');
+
+    app.use(express.static(path.join(__dirname, '.')));
+
+    app.listen(8000, () => {
+    	console.log('Server started on port 8000');
+    });
+    ```
+
+5. Save the file.
+6. Open your command prompt window and navigate to the directory where `server.js` is located.
+7. Type `node server.js` and press Enter. This will start the server on port 8000.
+8. Open your web browser and navigate to `http://localhost:8000/`. MetaMask should ask for your approval to connect to your website. Follow the steps and give your consent.
```

Comment:
Unfortunately, this would not work because of browser security restrictions. 
(I discovered that I had a very old question about this https://github.com/MetaMask/metamask-extension/issues/6162 :smile:)